app/controllers/api/base_controller.rb

```diff
@@ -133,7 +133,16 @@ def set_cache_headers
   end
 
   def disallow_unauthenticated_api_access?
-    ENV['DISALLOW_UNAUTHENTICATED_API_ACCESS'] == 'true' || Rails.configuration.x.whitelist_mode
+    (ENV['DISALLOW_UNAUTHENTICATED_API_ACCESS'] == 'true' || Rails.configuration.x.whitelist_mode) unless current_user
```

Comment:
```suggestion
    return false if current_user
    ENV['DISALLOW_UNAUTHENTICATED_API_ACCESS'].casecmp?('true' ) || Rails.configuration.x.whitelist_mode
```

Functionally the same, but I think it offers some readability improvements.

By explicitly returning in the first line you're saving some work. 

And `ENV['DISALLOW_UNAUTHENTICATED_API_ACCESS'].casecmp?('true' )` will return a boolean true if the ENV var is "True", "TRUE", "true" or any other combination. Does require Ruby 2.4 or later.

Should you be on < 2.4 I suggest 

```ruby
ENV['DISALLOW_UNAUTHENTICATED_API_ACCESS'].casecmp('true') == 0
```

Reply:
I agree with the reformat for clearer intention, though I don't want to change the semantics of the env var check here (which are from upstream). Other places that check env vars do so with case sensitive comparisons.